bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/st_tch_4f_ckm_13TeV_UL/st_tch_4f_ckm_width_variations/gridpack_generation_topwidth.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/bin/MadGraph5_aMCatNLO/gridpack_generation.sh b/bin/MadGraph5_aMCatNLO/gridpack_generation.sh
+index c8ebed5..134d629 100755
+--- a/bin/MadGraph5_aMCatNLO/gridpack_generation.sh
++++ b/bin/MadGraph5_aMCatNLO/gridpack_generation.sh
+@@ -471,6 +471,9 @@ make_gridpack () {
+       echo "copying custom params file"
+       cp $CARDSDIR/${name}_param_card.dat ./Cards/param_card.dat
+     fi
++
++    sed -i "s/decay \+6 \+.*/DECAY  6  ${TOPWIDTH}\n#  BR             NDA  ID1    ID2   ...\n1.000000e+00   2    24  5/gi" ./Cards/param_card.dat
++    sed -i "s/decay \+24 \+.*/#      PDG        Width\nDECAY  24   2.049946e+00\n#  BR             NDA  ID1    ID2   ...\n3.163393e-01   2    4  -3 # 0.648478474205\n3.163393e-01   2    2  -1 # 0.648478474205\n1.110098e-01   2    14  -13 # 0.227564079932\n1.110098e-01   2    12  -11 # 0.227564079932\n1.109285e-01   2    16  -15 # 0.227397411854\n1.690460e-02   2    2  -3 # 0.0346535195344\n1.690460e-02   2    4  -1 # 0.0346535195344\n5.602090e-04   2    4  -5 # 0.00114839807192\n3.802070e-06   2    2  -5 # 7.79403741826e-06/gi" ./Cards/param_card.dat
```

Comment:
Hi @swertz . Can you explain why you change these lines? What changes for the W?

Reply:
It seems MadSpin will not use the decay informaton from the banner unless ALL the widths and branching ratios of the particles for which decays are requested in the madspin_card are provided in the banner.
Since we have `decay W > all` in MadSpin, we also need to add the values for the W.
The W decay is indeed not modified, but this adds the branching ratios in the banner at the right place.

Reply:
nice finding